src/send/mod.rs

```diff
@@ -97,10 +103,17 @@ fn fail() -> data::Outcome<Email, ValidationError>
 }
 
 #[post("/send", format = "application/json", data = "<email>")]
-fn handler(email: Email) -> Result<Json<Value>, Error>
+fn handler(email: Email) -> Result<Json<Value>, Failure>
```

Comment:
Returning `Error` meant that all failures get treated as `500`. This lets us return a `429`.

Reply:
If `429` the appropriate code to return here?  IIUC, it's not that the auth-server itself has sent too many requests, but that there's a problem with the specific request being sent.  I wonder if a `400` and an `errno` would be a better fit here.

Reply:
I did raise this question in the feature doc fwiw, but didn't get any feedback there. I'll run through some of the considerations again quickly:

* I started off with `400` for this in the feature doc then I noticed SergioBenitez/Rocket#46, which talks about `400` only being appropriate for malformed payloads and `422` being correct for other validation errors. That started me thinking.

* Semantically, I think of `400` as an instruction to the caller that it must change the payload for the request to work. But that's not the case here, only some time needs to pass in order for the exact same payload to work fine if tried again. That does seem similar to `429` semantics, even if it's not a rate-limit in the conventional sense.

* Then I looked at the [MDN docs for `429`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/429) and the language there framed the error in terms of the user:

  > the user has sent too many requests in a given amount of time

  Which made me think it would be okay to consider this response as per-email address rather than a blanket-wide case for the auth server in general. Also the Amplitude API has been in my head recently and they behave in a similar way, returning `429` for specific values of `device_id`.

* On a philosophical level, doesn't "appropriate" here just mean whatever we choose to document for the API (within reason)? This is the point in my thinking where I opted to define it as `"code": 429, "errno": 103` and resolve the comment thread in the feature doc.

* On a practical level, since the implementation work started, I had some issues getting Rocket's error catchers working properly. I'm sure we'll figure those out easily when we go back to it but, in the meantime, we can't differentiate on `errno` anyway for now. So there's a separate argument for leaving this until we resolve #8, regardless, because the tests are stricter if we can differentiate by HTTP status instead.

None of that is strong disagreement btw, I just wanted to show how I reached this point. What is the case for `400` being the correct response?

Reply:
> I did raise this question in the feature doc fwiw, but didn't get any feedback there

Apologies for missing it there!

> On a philosophical level, doesn't "appropriate" here just mean whatever
> we choose to document for the API (within reason)?

Yes, yes it does.  I don't have any strong objection to the rationale you set out above.  My one concern would be if, in a future where this is generally available, we want to use 429 to tell the actual application server to back off.  I don't have a compellingly better proposal though, so happy to kick that down the road.